xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -848,6 +855,73 @@ bool CProcessorHD::IsBT2020Supported()
   return false;
 }
 
+bool CProcessorHD::IsHDR10PassthroughSupported()
+{
+  ComPtr<ID3D11VideoDevice> pVideoDevice;
+  ComPtr<ID3D11VideoProcessorEnumerator> pEnumerator;
+  ComPtr<ID3D11VideoProcessorEnumerator1> pEnumerator1;
+  ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
+
+  if (FAILED(pD3DDevice.As(&pVideoDevice)))
+  {
+    CLog::LogF(LOGERROR, "failed to get video device.");
+    return false;
+  }
+
+  D3D11_VIDEO_PROCESSOR_CONTENT_DESC desc = {};
+  desc.InputWidth = 3840;
+  desc.InputHeight = 2160;
+  desc.OutputWidth = 3840;
+  desc.OutputHeight = 2160;
+
+  if (FAILED(pVideoDevice->CreateVideoProcessorEnumerator(&desc, &pEnumerator)))
+  {
+    CLog::LogF(LOGERROR, "failed to create Video Enumerator.");
+    return false;
+  }
+
+  if (FAILED(pEnumerator.As(&pEnumerator1)))
+  {
+    CLog::LogF(LOGWARNING,
+               "ID3D11VideoProcessorEnumerator1 is not available. DXVA will not be used.");
+    return false;
+  }
+
+  const DXGI_FORMAT destFormat = DX::Windowing()->GetBackBuffer().GetFormat();
+  const DXGI_COLOR_SPACE_TYPE destColor = DX::Windowing()->UseLimitedColor()
+                                              ? DXGI_COLOR_SPACE_RGB_STUDIO_G2084_NONE_P2020
+                                              : DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+  BOOL supported = 0;
+
+  // Check if HDR10 (TOP LEFT) input color space is supported by video driver
+  HRESULT hr = pEnumerator1->CheckVideoProcessorFormatConversion(
+      DXGI_FORMAT_P010, DXGI_COLOR_SPACE_YCBCR_STUDIO_G2084_TOPLEFT_P2020, destFormat, destColor,
```

Comment:
Assuming P010 will work 99% of the time but I have a sample with 8 bit PQ+BT.2020
AMD doesn't support NV12 + PQ and BT.2020 as input, but Intel Gen 8 does and outputs correctly.
Pass the format NV12/P010 from GetWeight?

Reply:
I'm not sure of this: is supposed HDR10 is always P010